tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -114,4 +115,60 @@ var _ = Describe("odo devfile create command tests", func() {
 			helper.MatchAllInOutput(output, []string{expectedString})
 		})
 	})
+
+	Context("When executing odo create with devfile component and --downloadSource flag", func() {
+		It("should succesfully create the compoment and download the source", func() {
+			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+			contextDevfile := helper.CreateNewContext()
+			helper.Chdir(contextDevfile)
+			devfile := "devfile.yaml"
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", devfile), filepath.Join(contextDevfile, devfile))
+			// TODO: Check for devfile in non-default location
+			helper.CmdShouldPass("odo", "create", "nodejs", "--downloadSource", "--devfile", devfile)
+			expectedFiles := []string{"package.json", "package-lock.json", "README.MD", devfile}
+			Expect(helper.VerifyFilesExist(contextDevfile, expectedFiles)).To(Equal(true))
+			helper.DeleteDir(contextDevfile)
+		})
+	})
+
+	Context("When executing odo create with devfile component and --downloadSource flag with github type", func() {
+		It("should succesfully create the compoment and download the source", func() {
+			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+			contextDevfile := helper.CreateNewContext()
+			helper.Chdir(contextDevfile)
+			devfile := "devfile.yaml"
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", devfile), filepath.Join(contextDevfile, devfile))
+
+			err := helper.ReplaceDevfileField(devfile, "type", "github")
+			if err != nil {
+				log.Error("Could not replace the entry in the devfile: " + err.Error())
+			}
+			helper.CmdShouldPass("odo", "create", "nodejs", "--downloadSource", "--devfile", devfile)
+			expectedFiles := []string{"package.json", "package-lock.json", "README.MD", devfile}
+			Expect(helper.VerifyFilesExist(contextDevfile, expectedFiles)).To(Equal(true))
+			helper.DeleteDir(contextDevfile)
+		})
+	})
+
+	Context("When executing odo create with devfile component and --downloadSource flag with zip type", func() {
+		It("should create the compoment and download the source", func() {
+			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+			contextDevfile := helper.CreateNewContext()
+			helper.Chdir(contextDevfile)
+			devfile := "devfile.yaml"
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", devfile), filepath.Join(contextDevfile, devfile))
+			err := helper.ReplaceDevfileField(devfile, "location", "https://github.com/che-samples/web-nodejs-sample/archive/master.zip")
+			if err != nil {
+				log.Error("Could not replace the entry in the devfile: " + err.Error())
+			}
+			err = helper.ReplaceDevfileField(devfile, "type", "zip")
+			if err != nil {
+				log.Error("Could not replace the entry in the devfile: " + err.Error())
+			}
+			helper.CmdShouldPass("odo", "create", "nodejs", "--downloadSource", "--devfile", devfile, "--context", filepath.Join(context, "config.yaml"))
```

Comment:
FYI, currently we only support `--context` for s2i component, `--context` support for devfile component will be tacked by this issue https://github.com/openshift/odo/issues/2824. For we now we can only use helper.Chdir(<destination>) instead of using `--context`

Reply:
The issue is that we don't want to have the `config.yaml` and the `devfile.yaml` in the same folder, but we still need the experimental flag set which is stored in `config.yaml`? So if I don't specify context, we cannot read the preference and if I chdir with the two files my code fails.

Reply:
Why don't we want to have the config.yaml in the directory? 

FWIW, using the `--context` flag right now in odo for devfiles is pretty much a no-op so it should be fine to leave in.

Reply:
@EnriqueL8 you can check our devfile push test here https://github.com/openshift/odo/blob/master/tests/integration/devfile/cmd_devfile_push_test.go#L31

- It uses context directory to store `config.yaml` and set up the file location via configuring `GLOBALODOCONFIG` env var
- It creates a new project directory to store both template and `devfile.yaml`

With that change `config.yaml` and `devfile.yaml` are in different directory, and we can use `helper.Chdir(<destination>)` to switch to the  new project directory to do all the devfile related operations and we can read the preference from there as well.

Reply:
FYI, I just drop a PR https://github.com/openshift/odo/pull/2917 for `--context` support for devfile component